src/include/ipc4/header.h

```diff
@@ -0,0 +1,103 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ */
+
+/**
+ * \file include/ipc4/header.h
+ * \brief IPC4 gloabl definitions
+ */
+
+#ifndef __SOF_IPC4_HEADER_H__
+#define __SOF_IPC4_HEADER_H__
+
+#include <stdint.h>
+
+//! Message target, value of msg_tgt field.
+enum ipc4_message_target {
+	//! Global FW message
+	SOF_IPC4_MESSAGE_TARGET_FW_GEN_MSG = 0,
+	//! Module message
+	SOF_IPC4_MESSAGE_TARGET_MODULE_MSG = 1
+};
+
+//! Message direction, value of rsp field.
+enum ipc4_message_direction {
+	//! Request, Notification
+	SOF_IPC4_MESSAGE_DIR_MSG_REQUEST = 0,
+	//! Reply
+	SOF_IPC4_MESSAGE_DIR_MSG_REPLY = 1,
+};
+
+/*
+ * Global IPC4 message types - must fit into 5 bits.
+ */
+enum ipc4_message_type {
+	//! Boot Config.
+	SOF_IPC4_GLB_BOOT_CONFIG = 0,
+	//! ROM Control (directed to ROM).
+	SOF_IPC4_GLB_ROM_CONTROL = 1,
+	//! Execute IPC gateway command
+	SOF_IPC4_GLB_IPCGATEWAY_CMD = 2,
+	//! Create EDF task and run FreeRTOS instance in it
+	SOF_IPC4_GLB_START_RTOS_EDF_TASK = 3,
+	//! Stop FreeRTOS and delete its EDF task context
```

Comment:
are the references to FreeRTOS necessary? that doesn't seem very portable or future-proof.

Reply:
This is an ***exact copy*** of the interface being used, agree it' has many issues around portability and being too Intel specific. However it's a one device only - I will make this really clear in the comments.

Reply:
Removed and added a reserved place holder.